packages/ui/src/modal/buy/BuyModal.tsx

```diff
@@ -93,10 +95,34 @@ export function BuyModal({
     openState ? openState[0] : false,
     openState
   )
-  const { chain: activeChain } = useNetwork()
+
+  const { chains } = useNetwork()
+  const client = useReservoirClient()
+  const { switchNetworkAsync } = useSwitchNetwork()
+
+  const currentChain = client?.currentChain()
+
+  const modalChain = chainId
+    ? client?.chains.find(({ id }) => {
+        id === chainId
+      }) || currentChain
+    : currentChain
+
+  const walletChain = chains.find(({ id }) => {
+    modalChain?.id === id
+  })
+
+  const handleBuy = async (buyToken: () => void): Promise<void> => {
+    if (modalChain?.id !== currentChain?.id) {
```

Comment:
This should check the modalChain (currently configured chain on the modal) against the activeChain (the chain that the user is connected to in their wallet)